metriq_gym/qplatform/device.py

```diff
@@ -1,12 +1,336 @@
+from typing import cast, Dict, Any, Protocol
+import tempfile
+import json
+import uuid
+import os
+from abc import ABC, abstractmethod
 
 import networkx as nx
 from qbraid import QuantumDevice
 from qbraid.runtime import BraketDevice, QiskitBackend
 import rustworkx as rx
 
 
+class SimulatorAdapter(Protocol):
+    """Protocol defining the interface for simulator adapters."""
+    
+    def run_circuits(self, circuits, shots: int) -> Any:
+        """Run circuits on the simulator backend."""
+        ...
+    
+    def get_backend_info(self) -> Dict[str, Any]:
+        """Get backend information (num_qubits, basis_gates, etc.)."""
+        ...
+    
+    def get_version(self) -> str:
+        """Get simulator version."""
+        ...
+
+
+class QiskitAerAdapter:
+    """Adapter for Qiskit Aer simulators."""
+    
+    def __init__(self, method: str = "automatic"):
+        """Initialize Qiskit Aer adapter.
+        
+        Args:
+            method: Simulation method ('automatic', 'statevector', 'stabilizer', etc.)
+        """
+        try:
+            from qiskit_aer import AerSimulator
+            
+            if method == "automatic":
+                self._backend = AerSimulator()
+            else:
+                self._backend = AerSimulator(method=method)
+                
+            self.method = method
+            
+        except ImportError:
+            raise ImportError(
+                "Qiskit Aer is required for Qiskit simulators. "
+                "Install with: pip install qiskit-aer"
+            )
+    
+    def run_circuits(self, circuits, shots: int):
+        """Run circuits on Qiskit Aer."""
+        if isinstance(circuits, list):
+            results = []
+            for circuit in circuits:
+                result = self._backend.run(circuit, shots=shots).result()
+                results.append(result)
+            return results
+        else:
+            return [self._backend.run(circuits, shots=shots).result()]
+    
+    def get_backend_info(self) -> Dict[str, Any]:
+        """Get Qiskit backend information."""
+        config = self._backend.configuration()
+        return {
+            'num_qubits': getattr(config, 'n_qubits', 64),
+            'basis_gates': getattr(config, 'basis_gates', [
+                'u1', 'u2', 'u3', 'cx', 'id', 'x', 'y', 'z', 'h', 's', 't', 'sdg', 'tdg'
+            ]),
+            'coupling_map': getattr(config, 'coupling_map', None),
+            'method': self.method,
+            'backend_name': config.backend_name,
+        }
+    
+    def get_version(self) -> str:
+        """Get Qiskit Aer version."""
+        try:
+            import qiskit_aer
+            return qiskit_aer.__version__
+        except (ImportError, AttributeError):
+            return "unknown"
+
+
+# Registry of available simulator adapters
+SIMULATOR_ADAPTERS = {
+    "qiskit.aer": QiskitAerAdapter,
+}
+
+# Topology configurations
+TOPOLOGY_CONFIGS = {
+    "line": {
```

Comment:
I wonder if we want to `StrEnum` these topology types for the outer keys. Not necessary, but maybe a thought.